From 2f6a4358d970b83561144eba26de381b94bb93f2 Mon Sep 17 00:00:00 2001
From: Dmitry Muhomor <muhomor.dmitry@gmail.com>
Date: Wed, 29 May 2024 19:36:14 +0300
Subject: [PATCH] disable support for adoptable storage

Adoptable storage is not available on any of the currently supported devices.

Handling of adoptable storage would complicate the duress password implementation.
---
 core/java/android/os/storage/DiskInfo.java               | 3 +--
 .../java/com/android/server/StorageManagerService.java   | 9 ---------
 2 files changed, 1 insertion(+), 11 deletions(-)

diff --git a/core/java/android/os/storage/DiskInfo.java b/core/java/android/os/storage/DiskInfo.java
index d32928cbe..d51e20dce 100644
--- a/core/java/android/os/storage/DiskInfo.java
+++ b/core/java/android/os/storage/DiskInfo.java
@@ -46,7 +46,6 @@ public class DiskInfo implements Parcelable {
     public static final String EXTRA_VOLUME_COUNT =
             "android.os.storage.extra.VOLUME_COUNT";
 
-    public static final int FLAG_ADOPTABLE = 1 << 0;
     public static final int FLAG_DEFAULT_PRIMARY = 1 << 1;
     public static final int FLAG_SD = 1 << 2;
     public static final int FLAG_USB = 1 << 3;
@@ -136,7 +135,7 @@ public class DiskInfo implements Parcelable {
 
     @UnsupportedAppUsage
     public boolean isAdoptable() {
-        return (flags & FLAG_ADOPTABLE) != 0;
+        return false;
     }
 
     @UnsupportedAppUsage
diff --git a/services/core/java/com/android/server/StorageManagerService.java b/services/core/java/com/android/server/StorageManagerService.java
index 3689490d2..64751a391 100644
--- a/services/core/java/com/android/server/StorageManagerService.java
+++ b/services/core/java/com/android/server/StorageManagerService.java
@@ -1423,15 +1423,6 @@ class StorageManagerService extends IStorageManager.Stub
         @Override
         public void onDiskCreated(String diskId, int flags) {
             synchronized (mLock) {
-                final String value = SystemProperties.get(StorageManager.PROP_ADOPTABLE);
-                switch (value) {
-                    case "force_on":
-                        flags |= DiskInfo.FLAG_ADOPTABLE;
-                        break;
-                    case "force_off":
-                        flags &= ~DiskInfo.FLAG_ADOPTABLE;
-                        break;
-                }
                 mDisks.put(diskId, new DiskInfo(diskId, flags));
             }
         }
-- 
2.43.0

